### ISSUE TITLE
Простой пример с TypeScripit

### ISSUE THREAD

**Issue by IgorKulishov** [g1er/Andrew#18]
В данном примере мы рассмотрим простой пример построения вэб приложения.
Я уже добавил пример в репозитарий, так чтот тебе будет проще pull репозитарий (master branch) и перейти в папку typescript или посмотреть online на сайта: https://github.com/g1er/Andrew/tree/master/typescript

В данном примере мы добавили:
1. файл **index.html** 
Внутри самого файла добавили <script> тэг, который будет указывать на путь к откомпелированному файлу: js/app.js (файл после компелации);
2. внутри иходного файла app.ts мы добавим код с применением typescript class (классы TypeScript очень похожи на классы javascript ES6):
```
class Main {
    constructor() {
    }
    popUp() {
        alert('Hello World!');
    }
}
let main = new Main();
main.popUp();
```
3. все остальные файлы остаются без изменения.
4. после того как мы создали **index.html** с script тэгом указывающим на js/app.js и добивили код в файл app.ts необходимо осуществить компеляцию посредством нажатия "Ctrl+Shift+B" врезультате появится файл js/app.js.
5. далее надо перейти в браузере при помощи встроенного Miscrosoft протокола "file:///" и набрать в браузере "file:///C:/" далее нажимая на папки перейти и запустить файл **index.html**:
```
file:///C:/www/typescript-proj-git/Andrew/typescript/index.html
```
После запуска должен открыться alert с сообщением.

Если ты откроешь дебагер в браузере, откроешь третью закладку Sources , нажмешь Ctrl+O, и наберешь **app.ts**, то ты увидешь что откроется файл в формате **typescript**, как на приложенной картинке:
![screenshot](https://user-images.githubusercontent.com/6174568/39088288-febff8b0-457c-11e8-95cf-523b6800009b.png)

Это произошло за счет того что:
а) сгенерировался файл app.js.map, который и позволяет нам видеть исходный формат ts 
б) файл app.js.map сгенерировался за счет того что в конфигурационном файле tsconfig.json мы указали параметр "sourceMap": true

**Comment by g1er** [comment]
Я проделал все это, у меня вышла загвоздка. 
1) текст, который у тебя на первом скрине (создание класса Main) отличается от того, который у меня оказался:
```
function greeter(person) {
    return "Hello, " + person;
}

let user = "Jane User";

document.body.innerHTML = greeter(user);
```
теперь,
2) в моем примере, когда я запускаю файл index.html, у меня не выходит Hello, Jane User, потому что дебаггер пишет "Uncaught TypeError: Cannot set property 'innerHTML' of null at app.ts:7".
![default](https://user-images.githubusercontent.com/24845718/39415791-dbdc3a6c-4c60-11e8-9380-a4b12457a3c6.jpg)
Твой пример работает нормально, alert показывает Hello, Wolrd, а в моем случае скрипт не срабатывает. Сама по себе функция рабочая, я проверял. И я, если честно, не могу разобрать, что там не так

**Comment by g1er** [comment]
Да, причем это отображается только в файле TS, как видишь. В файле JS никаких ошибок дебаггер не обнаруживает. Тем не менее, скрипт JS не срабатывает

**Comment by IgorKulishov** [comment]
Ты можешь разрешить вопрос окружив document.body.innerHTML следующим условием:
```
function greeter(person) {
    return "Hello, " + person;
}

let user = "Jane User";

let greeterValue = greeter(user);
console.log(greeterValue);
if (document.body) {
    document.body.innerHTML = greeterValue;
}
```
Т.е. когда глобальная переменная document.body будет доступна внутри ф-ции она выполниться.
Это интересный вопрос и требует проработки, но как я понимаю:
1) глобальные методы alert, console.log доступны внутри браузера сразу
2) в то время как глобальный объект document.body должен подгрузиться

Прошу проверить работу, прислать вопрос и закомитеть изменения.
Мы можем оставить вопрос открытым для понимая причины такого поведения.

**Comment by IgorKulishov** [comment]
Кроме того можно сделать другой пример когда это тоже работает:
```
function greeter(person) {
    return "Hello, " + person;
}

let user = "Jane User";
let greeterValue = greeter(user);
 setTimeout( function() {
    document.body.innerHTML = greeterValue;
 }, 1);
```
Мы поставили задуржку на 1мс то что мы не почувствуем (на самом деле иногда используется трюк и ставят задержку на 0 мс - это должно работать).

**Comment by g1er** [comment]
Да, теперь все срабатывает как надо.
Интересная обработка, надо будет запомнить на будущее
